Add Flogger support

yiftizur opened the pull request.

alexejk commented:
From the 1.16.22 changelog I see:

> OLD-CRUFT: lombok.experimental.Builder and lombok.experimental.Value are deprecated remnants of when these features were still in experimental. They are now removed entirely. If your project is dependent on an older version of lombok which still has those; fret not, lombok still processes these annotations. It just no longer includes them in the jar.

This is interesting, as then we need to either have conditional based on Lombok version, or some approach.

@mplushnikov what's your take on breaking backwards compatibility with Lombok older than 1.16.22?

yiftizur commented:
@alexejk i've removed my style changes, and added tests for the new Logger class.
I have reverted my changes to the experimental classes following @mplushnikov changes.
Can you guys please take a look?
